src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java

```diff
@@ -0,0 +1,121 @@
+	public static final double ANGLE_TOLERANCE = 0.5;
+
+	// HARDWARE
+	private final CANSparkFlex launcherTopMotor;
+	private final CANSparkFlex launcherBottomMotor;
+	private final CANSparkFlex launcherAngleMotor;
+	private final RelativeEncoder launcherTopEncoder;
+	private final RelativeEncoder launcherBottomEncoder;
+	private final SparkAbsoluteEncoder launcherAngleEncoder;
+	private final SparkPIDController launcherAnglePIDController;
+
+	private final GenericEntry launcherSpeedEntry =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher Speed setpoint", SPEAKER_SHOOT_SPEED)
+					.withSize(2, 1)
+					.withWidget(BuiltInWidgets.kNumberSlider)
+					.withProperties(Map.of("Min", -1, "Max", 1))
+					.getEntry();
+
+	private final GenericEntry launcherAngleEntry =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher angle", getAngle())
+					.withSize(1, 1)
+					.withWidget(BuiltInWidgets.kTextView)
+					.getEntry();
+	private final GenericEntry launcherSpeed =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher Speed", 0)
+					.withSize(1, 1)
+					.withWidget(BuiltInWidgets.kTextView)
+					.getEntry();
+	// Constructor
+	public LauncherSubsystem() {
+
+		// MOTOR INSTANCE VARIBLES
+		// motors
+		launcherTopMotor = new CANSparkFlex(Hardware.LAUNCHER_TOP_MOTOR_ID, MotorType.kBrushless);
+		launcherBottomMotor = new CANSparkFlex(Hardware.LAUNCHER_BOTTOM_MOTOR_ID, MotorType.kBrushless);
+		launcherAngleMotor = new CANSparkFlex(Hardware.LAUNCHER_ANGLE_MOTOR_ID, MotorType.kBrushless);
+		// encoders
+		launcherTopEncoder = launcherTopMotor.getEncoder();
+		launcherBottomEncoder = launcherBottomMotor.getEncoder();
+		launcherAngleEncoder = launcherAngleMotor.getAbsoluteEncoder(Type.kDutyCycle);
+
+		// PID controllers
+		launcherAnglePIDController = launcherAngleMotor.getPIDController();
+		launcherAnglePIDController.setFeedbackDevice(launcherAngleEncoder);
+	}
+
+	public void configMotors() {
+		launcherTopMotor.restoreFactoryDefaults();
+		launcherBottomMotor.restoreFactoryDefaults();
+		launcherAngleMotor.restoreFactoryDefaults();
+		// idle mode (wow)
+		launcherTopMotor.setIdleMode(IdleMode.kCoast);
+		launcherBottomMotor.setIdleMode(IdleMode.kCoast);
+		launcherAngleMotor.setIdleMode(IdleMode.kBrake);
+		// inveritng the bottom motor lmao
+		launcherBottomMotor.setInverted(true);
+
+		// current limit
+		launcherTopMotor.setSmartCurrentLimit(20);
+		launcherBottomMotor.setSmartCurrentLimit(20);
+		launcherAngleMotor.setSmartCurrentLimit(20);
+
+		launcherTopMotor.burnFlash();
+		launcherBottomMotor.burnFlash();
+		launcherAngleMotor.burnFlash();
+	}
+
+	// stop specific motor method
+	public void stopLauncher() {
+		launcherTopMotor.stopMotor();
+		launcherBottomMotor.stopMotor();
+	}
+
+	public void launch() {
+		launcherTopMotor.set(launcherSpeedEntry.getDouble(SPEAKER_SHOOT_SPEED));
+		launcherBottomMotor.set(launcherSpeedEntry.getDouble(SPEAKER_SHOOT_SPEED));
```

Comment:
```suggestion
        double speed = launcherSpeedEntry.getDouble(SPEAKER_SHOOT_SPEED);
		launcherTopMotor.set(speed);
		launcherBottomMotor.set(speed);
```
This guarantees that the the motors are set to the same speed (in case the entry gets set to a different value between the calls).

Reply:
Okay, i did exactly that, tysm!